src/components/cl/basic/cl_basic_coll.c

```diff
@@ -7,26 +7,43 @@
 #include "cl_basic.h"
 #include "utils/ucc_coll_utils.h"
 
-ucc_status_t ucc_cl_basic_select_tl(ucc_base_coll_args_t *args,
+ucc_status_t ucc_cl_basic_select_tl(ucc_base_coll_args_t *coll_args,
                                     ucc_cl_basic_team_t *cl_basic_team,
                                     ucc_tl_team_t **tl_team)
 {
+    ucc_coll_args_t *args = &coll_args->args;
     /* If NCCL TL is available and both src and dst buffers are CUDA then
        choose NCCL TL for alltoall and alltoallv otherwise use UCP TL*/
     if (cl_basic_team->tl_nccl_team != NULL) {
-        switch (args->args.coll_type) {
+        switch (args->coll_type) {
+        case UCC_COLL_TYPE_ALLGATHER:
+        case UCC_COLL_TYPE_ALLREDUCE:
         case UCC_COLL_TYPE_ALLTOALL:
-            if ((args->args.dst.info.mem_type == UCC_MEMORY_TYPE_CUDA) &&
-                (UCC_IS_INPLACE(args->args) ||
-                 (args->args.src.info.mem_type == UCC_MEMORY_TYPE_CUDA))) {
+            if ((args->dst.info.mem_type == UCC_MEMORY_TYPE_CUDA) &&
+                (UCC_IS_INPLACE(*args) ||
+                 (args->src.info.mem_type == UCC_MEMORY_TYPE_CUDA))) {
+                *tl_team = cl_basic_team->tl_nccl_team;
+                return UCC_OK;
+            }
+            break;
+        case UCC_COLL_TYPE_ALLGATHERV:
+            if ((args->dst.info_v.mem_type == UCC_MEMORY_TYPE_CUDA) &&
+                (UCC_IS_INPLACE(*args) ||
+                 (args->src.info.mem_type == UCC_MEMORY_TYPE_CUDA))) {
                 *tl_team = cl_basic_team->tl_nccl_team;
                 return UCC_OK;
             }
             break;
         case UCC_COLL_TYPE_ALLTOALLV:
-            if ((args->args.dst.info_v.mem_type == UCC_MEMORY_TYPE_CUDA) &&
-                (UCC_IS_INPLACE(args->args) ||
-                 (args->args.src.info_v.mem_type == UCC_MEMORY_TYPE_CUDA))) {
+            if ((args->dst.info_v.mem_type == UCC_MEMORY_TYPE_CUDA) &&
+                (UCC_IS_INPLACE(*args) ||
+                 (args->src.info_v.mem_type == UCC_MEMORY_TYPE_CUDA))) {
+                *tl_team = cl_basic_team->tl_nccl_team;
+                return UCC_OK;
+            }
+            break;
+        case UCC_COLL_TYPE_BCAST:
+            if (args->src.info.mem_type == UCC_MEMORY_TYPE_CUDA) {
```

Comment:
@vspetrov @manjugv its not for this PR but is it make sense to build some static priority table per memory type and collective to select the team